apps/api/src/database/repositories/perps-repository.ts

```diff
@@ -0,0 +1,919 @@
+          data.accountSummary,
+        ).then((result) => ({ agentId: data.agentId, ...result })),
+      ),
+    );
+
+    batchResults.forEach((result, index) => {
+      const agent = batch[index];
+      if (!agent) {
+        // This should never happen, but handle it defensively
+        repositoryLogger.error(
+          `[PerpsRepository] Unexpected missing agent at index ${index}`,
+        );
+        return;
+      }
+
+      const agentId = agent.agentId;
+      if (result.status === "fulfilled") {
+        successful.push(result.value);
+      } else {
+        failed.push({
+          agentId,
+          error:
+            result.reason instanceof Error
+              ? result.reason
+              : new Error(String(result.reason)),
+        });
+      }
+    });
+  }
+
+  repositoryLogger.info(
+    `[PerpsRepository] Batch sync completed: ${successful.length} successful, ${failed.length} failed`,
+  );
+
+  return { successful, failed };
+}
+
+// =============================================================================
+// AGGREGATION QUERIES
+// =============================================================================
+
+/**
+ * Get perps competition statistics using efficient SQL aggregations
+ * @param competitionId Competition ID
+ * @returns Competition statistics
+ */
+async function getPerpsCompetitionStatsImpl(
+  competitionId: string,
+): Promise<PerpsCompetitionStats> {
+  try {
+    // Use a subquery to get the latest summary per agent, then aggregate
+    const statsQuery = dbRead
+      .selectDistinctOn([perpsAccountSummaries.agentId], {
+        agentId: perpsAccountSummaries.agentId,
+        totalEquity: perpsAccountSummaries.totalEquity,
+        totalVolume: perpsAccountSummaries.totalVolume,
+      })
+      .from(perpsAccountSummaries)
+      .where(eq(perpsAccountSummaries.competitionId, competitionId))
+      .orderBy(
+        perpsAccountSummaries.agentId,
+        desc(perpsAccountSummaries.timestamp),
+      )
+      .as("latest");
+
+    // Aggregate the latest summaries
+    const [stats] = await dbRead
+      .select({
+        totalAgents: drizzleCount(statsQuery.agentId),
+        totalVolume: sum(statsQuery.totalVolume),
+        averageEquity: avg(statsQuery.totalEquity),
+      })
+      .from(statsQuery);
+
+    // Get position count separately (can't join easily with the subquery)
+    const [positionStats] = await dbRead
+      .select({
+        totalPositions: drizzleCount(perpetualPositions.id),
+      })
+      .from(perpetualPositions)
+      .where(eq(perpetualPositions.competitionId, competitionId));
+
+    return {
+      totalAgents: stats?.totalAgents ?? 0,
+      totalPositions: positionStats?.totalPositions ?? 0,
+      totalVolume: Number(stats?.totalVolume ?? 0),
+      averageEquity: Number(stats?.averageEquity ?? 0),
+    };
+  } catch (error) {
+    repositoryLogger.error("Error in getPerpsCompetitionStats:", error);
+    throw error;
+  }
+}
+
```

Comment:
Implemented in #1219 : 

- getCompetitionLeaderboardSummaries (get latest account summaries for all agents in a competition, sorted for leaderboard)

- countBulkAgentPositionsInCompetitions (count positions for an agent across multiple comps in bulk)

https://github.com/recallnet/js-recall/pull/1219/files#diff-3852b6a981f940f317393ced9e5c9fef1141ef64cedcdf2c439abaffbcc28f56R812